src/java.base/unix/classes/sun/nio/fs/UnixFileSystemProvider.java

```diff
@@ -431,13 +431,14 @@ public void createDirectory(Path obj, FileAttribute<?>... attrs)
             dfd2 = dup(dfd1);
             dp = fdopendir(dfd1);
         } catch (UnixException x) {
+            IOException ioe = x.errno() == UnixConstants.ENOTDIR ?
+                new NotDirectoryException(dir.getPathForExceptionMessage()) :
+                x.asIOException(dir);
             if (dfd1 != -1)
-                UnixNativeDispatcher.close(dfd1);
+                ioe = UnixNativeDispatcher.close(dfd1, ioe, e -> null);
             if (dfd2 != -1)
-                UnixNativeDispatcher.close(dfd2);
-            if (x.errno() == UnixConstants.ENOTDIR)
-                throw new NotDirectoryException(dir.getPathForExceptionMessage());
-            x.rethrowAsIOException(dir);
+                ioe = UnixNativeDispatcher.close(dfd2, ioe, e -> null);
+            throw ioe;
```

Comment:
I think it would be simpler to just revert the change and just call close(dfd1, e -> null) and close(dfd2, e -> null).